internal/transport/keepalive_test.go

```diff
@@ -622,13 +622,13 @@ func (s) TestKeepaliveServerEnforcementWithObeyingClientWithRPC(t *testing.T) {
 func (s) TestKeepaliveServerEnforcementWithDormantKeepaliveOnClient(t *testing.T) {
 	serverConfig := &ServerConfig{
 		KeepalivePolicy: keepalive.EnforcementPolicy{
-			MinTime: 2 * time.Second,
+			MinTime: 100 * time.Millisecond,
 		},
 	}
 	clientOptions := ConnectOptions{
 		KeepaliveParams: keepalive.ClientParameters{
-			Time:    50 * time.Millisecond,
-			Timeout: 1 * time.Second,
+			Time:    10 * time.Millisecond,
```

Comment:
the comments in ClientParameters 

```go
type ClientParameters struct {
	// After a duration of this time if the client doesn't see any activity it
	// pings the server to see if the transport is still alive.
	// If set below 10s, a minimum value of 10s will be used instead.
	Time time.Duration // The current default value is infinity.
```

I'm confused. Do you think the are some overrides that are in place here?

Reply:
That check is performed when keepalive parameters are specified as dial options. See: https://github.com/grpc/grpc-go/blob/7437662fd5b853bd604201ddcc4f26496690ed97/dialoptions.go#L471

In this test, we are directly creating the transport and hence do not dial, and hence do no use dial options, and hence we don't do through that validation.

Reply:
👍